### DISCUSSION TITLE
Switch streaming input

### DISCUSSION THREAD

**Discussion by Aendrew87** [owncast/owncast#1575] [General Owncast discussion]
Hey,

i want to stream an existing mp4 file, at some point i would like to pause the stream and start at this point a few minutes later.

For streaming the file i use `ffmpeg -re -i input.mp4 -c copy -f flv rtmp://127.0.0.1:1935/live/<key>`

When i pause this process, for example with STRG+Z, a few seconds later the stream on the client side stops.  
It is also not possible to resume - using `fg` ends in 'Error writing trailer of rtmp://127.0.0.1:1935/live/<key>: Broken pipe'

I also thought about playin a dummy file during this time, but this did not work too.  
When starting the seconds ffmpeg process i get 'rtmp://127.0.0.1:1935/live/<key>: Input/output error'.

Has anybody a similar szenario and could help me out - thanks

**Comment by jeyemwey** [comment]
Hi, for starters, can you specify which version of Owncast you use? I'm not sure how ffmpeg deals with `Ctrl + Z`, but `Ctrl + C` should work. 

"What's different between Ctrl+Z and Ctrl+C in Unix command line?" https://superuser.com/a/262948

**Reply by Aendrew87** [reply]
Hey,

thanks for your reply - the installed version of owncast ist 0.0.10.

Ctrl + C will stop the process and result in `Exiting normally, received signal 2.`  
With `jobs` there is no paused process i could resume.

**Comment by aaronpk** [comment]
I would recommend looking into actual streaming solutions rather than doing it this way. You'll want to use something like OBS which can play back a file and encode it for a stream, and mix it and switch between cameras too. Starting and stopping the actual stream input is never going to give you a good result.

**Reply by Aendrew87** [reply]
Thanks, for that idea but obs neets a gui and my linux server only has she shell :|  
I tried obs on my windows client and it worked, i would prevere a solution on the server (linux) were owncast ist running.

**Comment by gabek** [comment]
Hi there,

Unfortunately I don't think what you're trying to do is going to work.  If Owncast detects that RTMP traffic has stopped it disconnects and ends the stream.  So there's no notion of temporarily stopping it and picking back up.

**Reply by Aendrew87** [reply]
Hi gabek,

with obs on windows i could "switch" to a different scene, something like that on linux (without the need of a gui) would be great.

**Reply by gabek** [reply]
The reason it works with OBS is because OBS never actually stops streaming, it's just changing the input content. You'd need a headless alternative.  Maybe something like https://github.com/ffplayout/ffplayout_engine ?

**Reply by Aendrew87** [reply]
Thanks, gabek
Replacing the "real stream" with a "dummy stream" for a short time is also possible... like advertisement on television...

I will check if that is possible with ffplayout - Thanks for the link.
